app/services/payment_gateway.rb

```diff
@@ -6,13 +6,25 @@ module PaymentGateway
   # in those environments, but it's annoying in development
   @available = true
   MUTEX = Mutex.new
+  logger = Rails.logger
+  logger.info "Initialised PaymentGateway logger"
 
   TASK = PeriodicTask.new(every: 5.seconds, run_immediately: !Rails.env.test?) do
     begin
       result = HTTParty.get ENV.fetch('PAYMENT_GATEWAY_PING_ENDPOINT')
-      MUTEX.synchronize { @available = result.success? }
-    rescue SystemCallError
-      MUTEX.synchronize { @available = false }
+      MUTEX.synchronize {
+        if result.success? != @available
+          logger.info "PaymentGateway: State change from old #{@available} to #{result.success?}"
+        end
+        @available = result.success?
+      }
+    rescue SystemCallError => e
+      MUTEX.synchronize {
+        logger.error "PaymentGateway: SystemCallError - #{e.class} #{e.message} " +
```

Comment:
this log line doesn't need to be inside the mutex - try to lock the absolute minimum that you need to lock